rts/System/Sound/OpenAL/Sound.cpp

```diff
@@ -529,6 +530,35 @@ void CSound::OpenLoopbackDevice(const std::string& deviceName)
 		return;
 	}
 
+	// 2.attempt: Workaround for SDL bug with some soundcards, returning more channels than we asked for (returned 8 in test case)
```

Comment:
"returned 8 in test case" -> vague, what test case? Describe the testcase briefly

Perhaps also describe what's wrong with extra channels, isn't it good that we got more than we asked to? How do the extras cause problems?

Reply:
Sorry, you can't know out of this PR alone.
But i made a very detailed description of this sound problem some weeks ago in dc:
https://discord.com/channels/549281623154229250/1125171668894101564

For any reason engine devs didn't allow 8 channels for SDL-audio.
I can't answer this question. I just saw the warning in log and figured out, that it's possible to force 2 channels and we are good. Maybe 8 channels can be used for 7.1 sound, but that would still be wrong, because i have only 1 stereo output device plugged in. It's a problem from sdl with some sound cards and this PR is a workaround.
See here how another project handled that situation the same way:
https://gist.github.com/thevaber/2b228214aeef447dc48e9408ee2e3674#file-soloud_sdl2_static-cpp-L112-L119

The engine stops the attempt to open this sdl-device when received channels are not 1 or 2. As an alternative the method OpenOpenALDevice is used then And that is opening the device by alcOpenDevice(nullptr).
The problem with alc is, that it doesn't follow changed default audio device.
So, when you have already loaded into the game and notice, that your default sound device in os is still on your speakers, but you need it on your headphones, you need to reload the whole engine to apply it.
But when opening sdl default device then it follows changes to default audio devices.